### ISSUE TITLE
GCC 13.2: warning: ISO C++ prohibits anonymous structs [-Wpedantic]

### ISSUE THREAD

**Issue by eliasdaler** [jrouwe/JoltPhysics#1011]
Hello. 

I'm getting the following warning [here](https://github.com/jrouwe/JoltPhysics/blob/8bd2c9a1babd6b2d0c66a721c0d1d87d257a7383/Jolt/Core/Color.h#L66).

```cpp
Jolt/Core/Color.h:67:17: warning: ISO C++ prohibits anonymous structs [-Wpedantic]
   67 |                 {
      |                 ^

```

Apparently anonymous structs in C++ is an MSVC extension. 😅

Commit: 8bd2c9a1babd6b2d0c66a721c0d1d87d257a7383

(I'm not sure what can be done about this to preserve the old API)

**Comment by eliasdaler** [comment]
Apparently GLM has a flag to disable anonymous struct usage in similar context like this: https://github.com/g-truc/glm/blob/4137519418a933e5863eea7c3ac53890ae7faf9d/glm/detail/type_vec3.hpp#L50

**Comment by jrouwe** [comment]
I think you must be turning this warning on specifically for your project. I compiled with gcc 13 and it doesn't trigger the warning (also [godbolt](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIAMykrgAyeAyYAHI%2BAEaYxBIA7KQADqgKhE4MHt6%2BASlpGQKh4VEssfFcSXaYDplCBEzEBNk%2BfoHVtQL1jQTFkTFxibYNTS257SO9Yf1lg5UAlLaoXsTI7Bzm/mHI3lgA1Cb%2BbsgKBPiCh9gmGgCC1zde6UZ7zGwKyUxre6foh1a39x2TAUCj2yAOCX%2Bdw0AE4vAxMvdrjCTJCkbDkV4wgR/GYAPoEZFEjGwlhyXF/YkoknU2lU%2Bk0hl02EAejZhzcew8ZWm6D2ADcxF5MHt%2BMQ9sRMOhSHtgJLGDLot4RYY%2BWJkggmOjaadiF4HNqiWiAYzYVjBAAOAlM4iU03Mh1Mp32qlslkcvYAJSlYM1DHCtENNPNBCthPtwDtjpdMejcdd7KOewA4vKGL7DAGg7SQ2GmdEo8747Giwn3UmLMqM/7MIGTQ7c9b7Vr/P9i%2B3SyXWYnOTdaBqmNWs/WjQARO3QlEJcetpHTv4cRa0TgAVl4fg4WlIqE4bms1m%2By1WIs2PFIBE0i8WAGsAv4AHT%2BJ/Pl8vgBs%2Bk4knXl%2B3nF4CggBo56XoscCwEgaAsMkdBxOQlBQTB9DxMAXC4nwdAEHEgEQNEv7RGEjQAJ6cGeBHMMQREAPLRNoNQXtwvBQWwghUQwtAkZuvBYNEXjAG4Yi0IBjGkFgLCGMA4hcaJeCSrU/KYMJW6YKoNReFhpG8Nita/rQeDRMQxEeFgv4EMQeAsJpixUAYwAKAAangmAAO5UckjCaTIggiGI7BSF58hKGov66Fw%2BgSSg%2B6WPo%2BmAZAiyoMkjgCMJAC0ADqgl7BlKlmUwvCoApxDmVgcUQIsHTJX4ECuGMfhhSE0ylOUeipOkVV1a1BRVX0zVzLYtb0XUkydWFlXDT0vUDBUww9KNs1NFNswVBVR5rBIS6rj%2B0k7hweyqBab6pW%2Bkiysg4JofeZh7BAuCECQBxmP4XDzLwDFaPMN53o%2Br6/U%2BH7Lhw36kBuW67QBQEgVxYEwIgIDLAQyTqfBECIbBxARKw6wHUdJ1ggYzwQGZ8LXq9QT4EQJV6Pw3miOI/k04FKjqNJoWkM5hnJFZn4cGuIO/rtVHqUjBB7KgVD7Ydx2ncA517Jd10QNySFxI9z1k%2B9V6kLeT4/X9r488DoMFf%2BtiQ5rn082Y21g6bFuLEV6TOJIQA%3D%3D%3D) doesn't) unless I specifically add `-Wpedantic`. In any case I disabled the warning inside Jolt as it compiles and works fine under gcc, clang and MSVC.
